Sources/TensorFlow/Optimizer.swift

```diff
@@ -355,3 +355,91 @@ public class AdaGrad<Model: Layer>: Optimizer
         update(&model.allDifferentiableVariables, along: direction)
     }
 }
+
```

Comment:
You need to add comments for the optimizer itself and a link to the paper that this is introduced in.